iceoryx_posh/test/moduletests/test_popo_server_port_common.hpp

```diff
@@ -0,0 +1,228 @@
+        iox::cxx::Ensures(!chunkSettingsResult.has_error());
+        auto& chunkSettings = chunkSettingsResult.value();
+
+        auto getChunkResult = m_memoryManager.getChunk(chunkSettings);
+        iox::cxx::Ensures(!getChunkResult.has_error());
+        return getChunkResult.value();
+    }
+
+    static constexpr uint64_t DUMMY_DATA{0U};
+
+    SharedChunk getChunkWithInitializedRequestHeaderAndData(const uint64_t data = DUMMY_DATA)
+    {
+        constexpr uint32_t USER_PAYLOAD_SIZE{sizeof(uint64_t)};
+        auto sharedChunk = getChunkFromMemoryManager(USER_PAYLOAD_SIZE, sizeof(RequestHeader));
+        new (sharedChunk.getChunkHeader()->userHeader())
+            RequestHeader(clientChunkQueueData.m_uniqueId, RpcBaseHeader::UNKNOWN_CLIENT_QUEUE_INDEX);
+        new (sharedChunk.getUserPayload()) uint64_t(data);
+        return sharedChunk;
+    }
+
+    uint64_t getRequestData(const RequestHeader* requestHeader)
+    {
+        iox::cxx::Ensures(requestHeader != nullptr && "requestHeader must not be a nullptr");
+        auto userPayload = ChunkHeader::fromUserHeader(requestHeader)->userPayload();
+        iox::cxx::Ensures(userPayload != nullptr && "userPayload must not be a nullptr");
+
+        return *static_cast<const uint64_t*>(userPayload);
+    }
+
+    /// @return true if all pushes succeed, false if a push failed and a chunk was lost
+    bool pushRequests(ChunkQueuePusher<ServerChunkQueueData_t>& chunkQueuePusher,
+                      uint64_t numberOfPushes,
+                      uint64_t requestDataBase = DUMMY_DATA)
+    {
+        for (uint64_t i = 0U; i < numberOfPushes; ++i)
+        {
+            auto sharedChunk = getChunkWithInitializedRequestHeaderAndData(requestDataBase + i);
+            if (!chunkQueuePusher.push(sharedChunk))
+            {
+                // this would actually be done by the ChunkDistributor from the ClientPort
+                chunkQueuePusher.lostAChunk();
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    void allocateResponseWithRequestHeaderAndThen(
+        SutServerPort& sut, std::function<void(const RequestHeader* const, ResponseHeader* const)> testFunction)
+    {
+        constexpr uint64_t USER_PAYLOAD_SIZE{8};
+        constexpr uint64_t USER_PAYLOAD_ALIGNMENT{8};
+
+        constexpr uint64_t NUMBER_OF_REQUESTS{1U};
+        pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+        auto requestResult = sut.portUser.getRequest();
+        ASSERT_FALSE(requestResult.has_error());
+        auto requestHeader = requestResult.value();
+
+        sut.portUser.allocateResponse(requestHeader, USER_PAYLOAD_SIZE, USER_PAYLOAD_ALIGNMENT)
+            .and_then([&](auto& responseHeader) { testFunction(requestHeader, responseHeader); })
+            .or_else([&](const auto& error) { FAIL() << "Expected ResponseHeader but got error: " << error; });
```

Comment:
Could you please use `GTEST_FAIL()` instead of `FAIL` in the whole test code base. According to `gtest.h` it was just defined since `FAIL` clashes sometimes with other libraries. Then it is also clear that you are using a gtest macro and not some custom defined one.

Reply:
Done for the files from this PR and created #1072 to replace it from the remaining code base since this PR is already quite big